File: 서지수/Week_7/P42577_전화번호목록.py
def solution(phone_book):
    answer = True
    phone_book.sort()
    # return answer

    # for i in range(len(phone_book)-1):
    #     for j in range(i+1, len(phone_book)):
    #         if phone_book[i] in phone_book[j] or phone_book[j] in phone_book[i]:
    #             answer=False
    
    for i in range(len(phone_book)-1):
        if len(phone_book[i])<=len(phone_book[i+1]):
            if phone_book[i] in phone_book[i+1]:
                if phone_book[i+1][:len(phone_book[i])]==phone_book[i]:
                   answer=False
        
    return answer

File: 길민지/Week_7/P42577_전화번호목록.java
import java.util.*;
class Solution {
    public boolean solution(String[] phone_book) {
       HashSet<String> set = new HashSet<>();
        for(String pb : phone_book){
            set.add(pb);
        }
       
        for(String pb : phone_book){
            for(int i=1; i<pb.length(); i++){
                if(set.contains(pb.substring(0, i))){
                    return false;
                }
            }
        }
        return true;
    }
}